client/components/NetworkList.vue

```diff
@@ -2,71 +2,185 @@
-				:prevent-on-filter="false"
-				:list="network.channels"
-				class="channels"
-				@change="onChannelSort"
-				@start="onDragStart"
-				@end="onDragEnd"
+		</div>
+		<div v-if="searchText" class="jump-to-results">
+			<div v-if="results.length">
+				<div
+					v-for="item in results"
+					:key="item.channel.id"
+					v-on="{mouseover: () => setActiveSearchItem(item.channel)}"
+					@click.prevent="selectResult"
+				>
+					<Channel
+						v-if="item.channel.type !== 'lobby'"
+						:channel="item.channel"
+						:network="item.network"
+						:active="item.channel === activeSearchItem"
+						:is-filtering="true"
+					/>
+					<NetworkLobby
+						v-else
+						:channel="item.channel"
+						:network="item.network"
+						:active="item.channel === activeSearchItem"
+						:is-filtering="true"
+					/>
+				</div>
+			</div>
+			<div v-else class="no-results">
+				No results found.
+			</div>
+		</div>
+		<Draggable
+			v-else
+			:list="networks"
+			:filter="isCurrentlyInTouch"
+			:prevent-on-filter="false"
+			handle=".channel-list-item[data-type='lobby']"
+			draggable=".network"
+			ghost-class="ui-sortable-ghost"
+			drag-class="ui-sortable-dragged"
+			group="networks"
+			class="networks"
+			@change="onNetworkSort"
+			@start="onDragStart"
+			@end="onDragEnd"
+		>
+			<div
+				v-for="network in networks"
+				:id="'network-' + network.uuid"
+				:key="network.uuid"
+				:class="{
+					collapsed: network.isCollapsed,
+					'not-connected': !network.status.connected,
+					'not-secure': !network.status.secure,
+				}"
+				class="network"
+				role="region"
 			>
-				<Channel
-					v-for="(channel, index) in network.channels"
-					v-if="index > 0"
-					:key="channel.id"
-					:channel="channel"
+				<NetworkLobby
 					:network="network"
+					:is-join-channel-shown="network.isJoinChannelShown"
+					:active="activeChannel && network.channels[0] === activeChannel.channel"
+					@toggleJoinChannel="network.isJoinChannelShown = !network.isJoinChannelShown"
 				/>
-			</Draggable>
-		</div>
-	</Draggable>
+				<JoinChannel
+					v-if="network.isJoinChannelShown"
+					:network="network"
+					:channel="network.channels[0]"
+					@toggleJoinChannel="network.isJoinChannelShown = !network.isJoinChannelShown"
+				/>
+
+				<Draggable
+					draggable=".channel-list-item"
+					ghost-class="ui-sortable-ghost"
+					drag-class="ui-sortable-dragged"
+					:group="network.uuid"
+					:filter="isCurrentlyInTouch"
+					:prevent-on-filter="false"
+					:list="network.channels"
+					class="channels"
+					@change="onChannelSort"
+					@start="onDragStart"
+					@end="onDragEnd"
+				>
+					<Channel
+						v-for="(channel, index) in network.channels"
+						v-if="index > 0"
+						:key="channel.id"
+						:channel="channel"
+						:network="network"
+						:active="activeChannel && channel === activeChannel.channel"
+					/>
+				</Draggable>
+			</div>
+		</Draggable>
+	</div>
 </template>
 
+<style>
+.jump-to-input {
```

Comment:
If we tweak the background to be non transparent on the input, we can make this sticky. What do you reckon?

```css
	position: sticky;
	top: 8px;
	z-index: 1;
```

Reply:
Not sure, this seems to cause issues with the magnifying glass icon positioning. Maybe we can test this out later and do it if it seems nice?

Reply:
Okay.